src/index.js

```diff
@@ -138,19 +138,36 @@ export const fuzzArrayExpression = f =>
   ap(Shift.ArrayExpression, {"elements": many(opt(choose(fuzzExpression, fuzzSpreadElement)))}, f);
 
 export const fuzzArrowExpression = (f = new FuzzerState) => {
-  let isConsise = f.rng.nextBoolean();
   let params, body;
-  if (!isConsise) {
+  let isAsync = f.rng.nextBoolean();
+  let isConcise = f.rng.nextBoolean();
+  // Because of the cover grammar we can't have an `await` identifier in the parameter list of an arrow function if we are in an async context
+  let outerForbidsAwait = !f.allowAwaitIdentifier;
+  if (!isConcise) {
     let {directives, hasStrictDirective} = fuzzDirectives(f);
-    f = f.enterFunction({isArrow: true, hasStrictDirective});
+    f = f.enterFunction({isArrow: true, isAsync, hasStrictDirective});
+    let oldAllowAwaitIdentifier = f.allowAwaitIdentifier;
+    if (outerForbidsAwait) {
+      f.allowAwaitIdentifier = false;
+    }
     params = fuzzFormalParameters(f, {hasStrictDirective});
+    if (outerForbidsAwait) {
```

Comment:
This guard isn't necessary, but it may be okay to leave it if you think it reads easier.

Reply:
Yeah, I prefer the guard, I think.